File: 139.cpp
#include<cmath>
#include<iostream>
#include<boost/math/common_factor.hpp>

int main(){

    int cnt = 0;
    int lim = 1e8;
    int a=0,b=0,c=0;

    int mxvl = 10000;

    for(int m = 1; m < mxvl; ++m)
    {
        for(int n = 1; n < mxvl; ++n)
        {
            a = m*m-n*n;
            b = 2*m*n;
            c = m*m+n*n;

            if(a+b+c < lim && c%(b-a)==0 && boost::math::gcd(a,b)==1 
                && boost::math::gcd(a,c)==1 && boost::math::gcd(b,c)==1)
            {
                cnt += lim/(a+b+c);
            }
        }
    }

    std::cout << cnt << std::endl;
    return 0;
}

File: 139.py
#! /usr/bin/python env

from math import *
from numpy import *

cnt = 0
lim = 1e8
eps = 1e-8
mxvl = 10000; # go to 10000 ultimately 

bsetrp = array([[ (u*u+2*u*v, 2*v*v+2*u*v, u*u+2*v*v+2*u*v) for u in xrange(1,mxvl)]\
        for v in xrange(1,mxvl)])

bsetrp = bsetrp.flatten()
bsetrp = split(bsetrp, len(bsetrp)/3)

for val in bsetrp:
    if(sum(val)<lim and val[2]%(val[1]-val[0])==0):
        cnt += 1

print cnt
